src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1048,6 +1082,29 @@ static int uct_ib_mlx5_is_xgvmi_alias_supported(struct ibv_context *ctx)
             UCT_IB_MLX5_HCA_CAPS_2_ALLOWED_OBJ_FOR_OTHER_VHCA_ACCESS_MKEY);
 }
 
+static int uct_ib_mlx5_mkey_by_name_reserve_set(uct_ib_mlx5_md_t *md, void *cap)
+{
+    int log_size;
+
+    ucs_assert(md->super.mkey_by_name_reserve.size == 0);
+
+    if (cap == NULL) {
```

Comment:
better check cap2 exists outside this function

Reply:
fixed, also for xgvmi case.